packages/frames.js/src/getFrameHtml.ts

```diff
@@ -54,13 +54,11 @@ export function getFrameHtmlHead(frame: Frame): string {
       : "",
     ...(frame.buttons?.flatMap((button, index) => [
       `<meta name="fc:frame:button:${index + 1}" content="${button.label}"/>`,
-      button.action
-        ? `<meta name="fc:frame:button:${index + 1}:action" content="${button.action}"/>`
-        : "",
+      `<meta name="fc:frame:button:${index + 1}:action" content="${button.action}"/>`,
       button.target
         ? `<meta name="fc:frame:button:${index + 1}:target" content="${button.target}"/>`
         : "",
-      button.post_url
+      button.action === "tx" && button.post_url
```

Comment:
i don't think this check should be added - post_url isn't limited to tx buttons by the spec (even though it's only useful for post buttons right now)

Reply:
okay got it, separate issue then